libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -980,6 +985,11 @@ void AP_Logger_File::io_timer(void)
         write_fd_semaphore.give();
         return;
     }
+
+#if AP_FILESYSTEM_LITTLEFS_ENABLED && CONFIG_HAL_BOARD != HAL_BOARD_SITL
+    bool sync_block = AP::littlefs().sync_block(_write_fd, _write_offset, nbytes);
```

Comment:
Could be done later, but it would be nicer to have a `bytes_until_sync` method on every filesystem, coupled with some logic to sync after a minimum number of bytes logged.

I would prefer not to add another singleton.